src/web/app/components/course-edit-form/course-edit-form.component.spec.ts

```diff
@@ -215,4 +215,135 @@ describe('CourseEditFormComponent', () => {
     expect(mockModalRef.componentInstance.activeCourses[0]).toEqual(testCourse);
     expect(mockModalRef.componentInstance.courseToFeedbackSession[testCourse.courseId]).toEqual([testFeedbackSession]);
   });
+
+  it('should set isEditing to true when editModel exists', () => {
+    component.editModel = DEFAULT_COURSE_EDIT_FORM_MODEL();
+
+    component.setIsEditing(true);
+
+    expect(component.editModel?.isEditing).toBe(true);
+  });
+
+  it('should set isEditing to false when editModel exists', () => {
+    component.editModel = DEFAULT_COURSE_EDIT_FORM_MODEL();
+
+    component.setIsEditing(false);
+
+    expect(component.editModel?.isEditing).toBe(false);
+  });
+
+  it('should not modify isEditing when editModel is undefined', () => {
+    component.editModel = undefined;
+
+    component.setIsEditing(true);
+
+    expect(component.editModel).toBeUndefined();
+  });
+
+  it('should update institutes when addModel is defined', () => {
+    component.addModel = DEFAULT_COURSE_ADD_FORM_MODEL() as CourseAddFormModel;
+    component.addModel.allCourses = [testCourse];
+    component.addModel.course = testCourse;
+
+    component.updateInstitutes();
+
+    expect(component.addModel.institutes).toEqual([validInstitute]);
+    expect(component.addModel.course.institute).toBe(validInstitute);
+  });
+
+  it('should not update institutes when addModel is not defined', () => {
+    component.addModel = undefined;
+
+    component.updateInstitutes();
+
+    expect(component.addModel).toBeUndefined();
+  });
+
+  it('should set the course timeZone when not in display-only mode', () => {
+    component.isDisplayOnly = false;
+
+    component.detectTimezoneHandler();
+
+    expect(component.model.course.timeZone).toBe(testTimeZone);
+  });
+
+  it('should not set the course timeZone when in display-only mode', () => {
+    component.isDisplayOnly = true;
+
+    component.detectTimezoneHandler();
+
+    expect(component.model.course.timeZone).not.toBe(testTimeZone);
+  });
+
+  it('should do nothing when in display-only mode', () => {
+    component.isDisplayOnly = true;
+    const emitSpy = jest.spyOn(component.updateCourseEvent, 'emit');
+
```

Comment:
we should also check for createCourseEvent here

Reply:
Okay will add test cases for that.